tutorial-1-translating-with-jobs/java/src/main/java/com/smartling/samples/jobstutorial/ProcessJob.java

```diff
@@ -0,0 +1,198 @@
+import com.smartling.api.v2.client.DefaultClientConfiguration;
+import com.smartling.api.v2.client.auth.Authenticator;
+import com.smartling.api.v2.client.auth.BearerAuthSecretFilter;
+import com.smartling.api.v2.client.exception.RestApiRuntimeException;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+
+public class ProcessJob
+{
+    private static final List<String> LOCALES = new ArrayList<>(Arrays.asList("fr-FR"));
+    private static final List<String> FILE_LIST = new ArrayList<>(Arrays.asList("site-navigation.json","products.json"));
+    private static final String FILE_TYPE = "json";
+
+    public static void main( String[] args ) throws Exception
+    {
+        // Read command-line arguments
+        if (args.length != 1) {
+            System.out.println("Usage: mvn exec:java@run");
+            System.out.println("(See pom.xml to configure parameters.)");
+            System.exit(1);
+        }
+        String jobName = args[0];
+
+        // Read authentication credentials from environment
+        String userId = System.getenv("DEV_USER_IDENTIFIER");
+        String userSecret = System.getenv("DEV_USER_SECRET");
+        String projectId = System.getenv("DEV_PROJECT_ID");
+        if (userId == null || userSecret == null || projectId == null) {
+            System.out.println("Missing environment variables. Did you run setenv?");
+            System.exit(1);
+        }
+
+        // Create factory for building API clients
+        DefaultClientConfiguration clientConfiguration = DefaultClientConfiguration.builder().build();
+        ClientFactory clientFactory = new ClientFactory();
+
+        // Create the authenticator to be used by subsequent API calls
+        AuthenticationApi authenticationApi = new AuthenticationApiFactory(clientFactory).buildApi(clientConfiguration);
+        Authenticator authenticator = new Authenticator(userId, userSecret, authenticationApi);
+
+        try {
+            System.out.println();
+
+            // create job
+            System.out.println("Calling 'create job' endpoint...");
+            TranslationJobsApi jobsApi = new TranslationJobsApiFactory(clientFactory)
+                    .buildApi(new BearerAuthSecretFilter(authenticator), clientConfiguration);
+            String jobUid = createJob(jobsApi, projectId, jobName);
+            System.out.println("Job created. Job ID: " + jobUid);
+
+            // create job batch
+            System.out.println("Calling 'create job batch' endpoint...");
+            JobBatchesApi jobBatchesApi = new JobBatchesApiFactory(clientFactory)
+                    .buildApi(new BearerAuthSecretFilter(authenticator), clientConfiguration);
+            String batchUid = createJobBatch(jobBatchesApi, projectId, jobUid);
```

Comment:
I would recommend:
- Create unique fileUris from `FILE_LIST` only once right above this string. I don't care about performance or code reuse. For some reasons the current implementation, where `createJobBatch` implicitly generates fileUris, confused me. Also it will be more close to real code when you have fileUri and work with it.
- Explicitly pass new unique fileUris into `createJobBatch`

Reply:
In addition to your suggestions, would it be clearer if I moved all the API-creation (jobsAPI = new Factory() etc) calls to the top where the authenticator is created? I guess that would reflect typical usage better in that these things only need to be created once?

Reply:
Have made the suggested changes.